android/app/src/main/java/app/eduroam/geteduroam/Route.kt

```diff
@@ -144,15 +159,20 @@ sealed class Route(val route: String) {
             return try {
                 val provider = configParser.parse(bytes)
                 inputStream.close()
-                "${BASE_URI}/${encodeArguments(null, provider)}"
+                "${BASE_URI}/${encodeArguments(ConfigSource.File, null, null, provider)}"
             } catch (ex: Exception) {
                 Timber.w(ex, "Could not parse file opened!")
                 null
             }
         }
 
-
-        fun encodeArguments(organizationId: String?, eapIdentityProviderList: EAPIdentityProviderList): String {
+        fun encodeArguments(
+            source: ConfigSource,
+            organizationId: String?,
+            organizationName: String?,
+            eapIdentityProviderList: EAPIdentityProviderList
+        ): String {
+            // TODO use type-safe navigation because this is something crashing, and we don't know why
```

Comment:
No, it is actually un-encoded JSON inside the URL, and the navigation route parser can't handle it. I have it already working with type-safe navigation on another branch, but even then I had to URL encode my JSON, otherwise it wouldn't work... So the new type safe solution isn't fully typesafe, it still converts to string routes :/